### ISSUE TITLE
dnsdist: Consider using BPF_MAP_TYPE_LPM_TRIE for range matching

### ISSUE THREAD

**Issue by rgacogne** [PowerDNS/pdns#9690]
- Program: dnsdist
 - Issue type: Feature request

### Short description
Linux supports the `BPF_MAP_TYPE_LPM_TRIE` type since 4.11, it would be nice to take advantage of that (when available) to offer eBPF range blacklisting in dnsdist.

### Usecase

Block a network range instead of a single address.

### Description

```
bpf = newBPFFilter(1024, 1024, 1024)
setDefaultBPFFilter(bpf)
bpf:blockRange("192.0.2.1/24")
```

**Comment by Y7n05h** [comment]
I will try to finish this future request next.

**Comment by rgacogne** [comment]
Implemented in #11526.
